### PR TITLE
Add separate document describing host probing logic

### PR DESCRIPTION
**Pull request by vitek-karas**
Describes in detail how the host does probing for assets.
Replace the existing short description of this in the shared store document.
Fix some related comments in the code.

- Please add a description for changes you are making.
- If there is an issue related to this PR, please add a reference to it.
- If this PR should not run tests please add say skip_ci_please in this description replacing the underscores with spaces.

### CODE REVIEW

**Comment by vitek-karas** [comment]
@dotnet-bot test Linux arm Release Build

File: Documentation/design-docs/host-probing.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  Given the base servicing path, the probing paths are
+    * Servicing NI probe path `<servicing base>/|arch|` - this is used only for `runtime` assets
+    * Servicing normal probe path `<servicing base>/pkgs` - this is used for all assets
+
+* The application (or framework if we're resolving framework assets) directory
+* Framework directories  
+  If the app (or framework) has dependencies on frameworks, these frameworks are used as probing paths.  
+  The order is from the higher level framework to lower level framework. The app is considered the highest level, it direct dependencies are next and so on.  
+  For assets from frameworks, only that framework and lower level frameworks are considered.
```
Comment (steveharter) [inline Documentation/design-docs/host-probing.md]: You may want to add a line about multilevel lookup where on Windows we always look in the global location even if running a copy of the runtime somewhere else

File: Documentation/design-docs/host-probing.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+The library relative path in this case is `newtonsoft.json/11.0.2` and the asset relative path is `lib/netstandard2.0/Newtonsoft.Json.dll`. So the goal of the probing logic is to find the `Newtonsoft.Json.dll` file using the above relative paths.
+
+## Probing
+The probing itself is done by going over a list of probing paths, which are ordered according to their priority. For each path, the host will append the relative parts of the path as per above and see if the file actually exists on the disk.  
+If the file is found, the probing is done, and the full path just resolved is stored.  
+If the file is not found, the probing continues with the next path on the list.
+If all paths are tried and the asset is still not found this is reported as an error (with the exception of app's `.deps.json` asset, in which case it's ignored).
+
+## Probing paths
```
Comment (steveharter) [inline Documentation/design-docs/host-probing.md]: Clarify framework-dependent vs. self-contained?

File: Documentation/design-docs/host-probing.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    * Otherwise (Linux/Mac) `$CORE_SERVICING`  
+
+  Given the base servicing path, the probing paths are
+    * Servicing NI probe path `<servicing base>/|arch|` - this is used only for `runtime` assets
+    * Servicing normal probe path `<servicing base>/pkgs` - this is used for all assets
+
+* The application (or framework if we're resolving framework assets) directory
+* Framework directories  
+  If the app (or framework) has dependencies on frameworks, these frameworks are used as probing paths.  
+  The order is from the higher level framework to lower level framework. The app is considered the highest level, it direct dependencies are next and so on.
```
Comment (steveharter) [inline Documentation/design-docs/host-probing.md]: Also when determining a probe path for a framework, higher level frameworks (and the app) are not considered

Comment (vitek-karas) [reply Documentation/design-docs/host-probing.md]: This is described in this line:
> For assets from frameworks, only that framework and lower level frameworks are considered.  
Which is pretty much exactly what you're asking for - isn't it?

File: Documentation/design-docs/host-probing.md (hunk 2 above)
Comment (vitek-karas) [reply Documentation/design-docs/host-probing.md]: Not exactly sure what you're asking for. I added a paragraph about this at the very end... but it's also already mentioned in the part about Framework directories.

**Pull request merged**